packages/client/prax.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * Prax is the reference implementation of browser-local Penumbra services. This
+ * module provides tools for connecting to Prax specifically, and a few
+ * additional conveniences.
+ */
+
+import type { JsonValue, ServiceType } from '@bufbuild/protobuf';
+import type { Transport } from '@connectrpc/connect';
+import { createPromiseClient } from '@connectrpc/connect';
+import { createChannelTransport } from '@penumbra-zone/transport-dom/create';
+import { jsonOptions } from '@penumbra-zone/types/registry';
+import { PenumbraSymbol } from './global';
+
+export class PraxNotAvailableError extends Error {}
+export class PraxNotConnectedError extends Error {}
+export class PraxManifestError extends Error {}
+
+export const getPraxPort = async () =>
+  window[PenumbraSymbol]?.['chrome-extension://lkpmkhpnhknhmibgnmmhdhgdilepfghe']?.connect()!;
+
+export const requestPraxConnection = async () =>
+  window[PenumbraSymbol]?.['chrome-extension://lkpmkhpnhknhmibgnmmhdhgdilepfghe']?.request();
+
+export const isPraxConnected = () =>
+  Boolean(
+    window[PenumbraSymbol]?.['chrome-extension://lkpmkhpnhknhmibgnmmhdhgdilepfghe']?.isConnected(),
+  );
+
+export const isPraxConnectedTimeout = (timeout: number) =>
+  new Promise<boolean>(resolve => {
+    if (
+      window[PenumbraSymbol]?.['chrome-extension://lkpmkhpnhknhmibgnmmhdhgdilepfghe']?.isConnected()
+    )
+      resolve(true);
```

Comment:
The conditional is so long, that without the brackets, I couldn't tell where it ended. Can we add them here? Or can you lift the statement out of the conditional and do:

```typescript
    const isConnected = window[PenumbraSymbol]?.['chrome-extension://lkpmkhpnhknhmibgnmmhdhgdilepfghe']?.isConnected();
    if (isConnected) resolve(true);
```